<!doctype html>
<!--
  Minimal Mistakes Jekyll Theme 4.24.0 by Michael Rose
  Copyright 2013-2020 Michael Rose - mademistakes.com | @mmistakes
  Free for personal and commercial use under the MIT license
  https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE
-->
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin _includes/seo.html --><title>Contents - Biological Modeling: A Free Online Course</title>
<meta name="description" content="Table of contents for Biological Modeling, a free online course.">



<meta property="og:type" content="website">
<meta property="og:locale" content="en_US">
<meta property="og:site_name" content="Biological Modeling: A Free Online Course">
<meta property="og:title" content="Contents">
<meta property="og:url" content="http://localhost:4000/contents/">


  <meta property="og:description" content="Table of contents for Biological Modeling, a free online course.">












<link rel="canonical" href="http://localhost:4000/contents/">




<script type="application/ld+json">
  {
    "@context": "https://schema.org",
    
      "@type": "Person",
      "name": null,
      "url": "http://localhost:4000/"
    
  }
</script>







<!-- end _includes/seo.html -->


<!-- https://t.co/dKP3o1e -->
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="/assets/css/main.css">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5/css/all.min.css">



<!--SEO tagging -->
<!-- Begin Jekyll SEO tag v2.6.1 -->
<title>Contents | Biological Modeling: A Free Online Course</title>
<meta name="generator" content="Jekyll v3.9.0" />
<meta property="og:title" content="Contents" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="Table of contents for Biological Modeling, a free online course." />
<meta property="og:description" content="Table of contents for Biological Modeling, a free online course." />
<link rel="canonical" href="http://localhost:4000/contents/" />
<meta property="og:url" content="http://localhost:4000/contents/" />
<meta property="og:site_name" content="Biological Modeling: A Free Online Course" />
<script type="application/ld+json">
{"description":"Table of contents for Biological Modeling, a free online course.","@type":"WebPage","url":"http://localhost:4000/contents/","headline":"Contents","@context":"https://schema.org"}</script>
<!-- End Jekyll SEO tag -->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-173522486-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-173522486-1');
</script>

    <!-- start custom head snippets -->

<!-- insert favicons. use https://realfavicongenerator.net/ -->

<!-- end custom head snippets -->

  </head>

  <body class="layout--single wide">
    <nav class="skip-links">
  <ul>
    <li><a href="#site-nav" class="screen-reader-shortcut">Skip to primary navigation</a></li>
    <li><a href="#main" class="screen-reader-shortcut">Skip to content</a></li>
    <li><a href="#footer" class="screen-reader-shortcut">Skip to footer</a></li>
  </ul>
</nav>

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->

    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
        <a class="site-title" href="/">Biological Modeling: A Free Online Course</a>
        <ul class="visible-links">
                
                
                    <li class="masthead__menu-item">
                        <a href="/meet-the-team/">Course Team</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/contact/">Contact Us</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/contents/">Contents</a>
                    </li>
                
            
                
                
                    <li class="dropdown ">
                        <a class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Take the Course    <i class="fa fa-caret-down fa-sm" aria-hidden="true"></i><span class="caret"></span></a>
                        <ul class="dropdown-content">
                            
                                <li>
                                    <a href="/prologue/">    <b>Prologue:</b> an introduction to biological modeling via random walks and Turing patterns</a>
                                </li>
                            
                                <li>
                                    <a href="/motifs/home">    <b>Module 1:</b> Finding motifs in transcription factor networks</a>
                                </li>
                            
                                <li>
                                    <a href="/chemotaxis/home">    <b>Module 2:</b> Unpacking E. coli's genius exploration algorithm</a>
                                </li>
                            
                                <li>
                                    <a href="/coronavirus/home">    <b>Module 3:</b> Analyzing the coronavirus spike protein</a>
                                </li>
                            
                                <li>
                                    <a href="">    <b>Module 4:</b> coming soon!</a>
                                </li>
                            
                        </ul>
                    </li>
                
            
        </ul>
        
        <button class="search__toggle" type="button">
          <svg class="icon" width="16" height="16" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15.99 16">
            <path d="M15.5,13.12L13.19,10.8a1.69,1.69,0,0,0-1.28-.55l-0.06-.06A6.5,6.5,0,0,0,5.77,0,6.5,6.5,0,0,0,2.46,11.59a6.47,6.47,0,0,0,7.74.26l0.05,0.05a1.65,1.65,0,0,0,.5,1.24l2.38,2.38A1.68,1.68,0,0,0,15.5,13.12ZM6.4,2A4.41,4.41,0,1,1,2,6.4,4.43,4.43,0,0,1,6.4,2Z" transform="translate(-.01)"></path>
          </svg>
        </button>
        
        <button class="greedy-nav__toggle hidden" type="button">
          <span class="visually-hidden">Toggle Menu</span>
          <div class="navicon"></div>
        </button>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>


    <div class="initial-content">
      



<div id="main" role="main">

  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>

  


  <article class="page" itemscope itemtype="https://schema.org/CreativeWork">
    <meta itemprop="headline" content="Contents">
    
    
    

    <div class="page__inner-wrap">
      
        <header>
          <h1 id="page-title" class="page__title" itemprop="headline">Contents
</h1>
          
        </header>
      

      <section class="page__content" itemprop="text">
        
        <div>
<script async="" src="https://cse.google.com/cse.js?cx=f517f335d138257dd"></script>
<div class="gcse-search"></div>
</div>

<p>This online course is divided into modules. Each module covers a single biological topic (e.g., “Analyzing the structure of the coronavirus spike protein”) and introduces all of the modeling topics needed to address that topic from first principles.</p>

<p>Each module has a main narrative that can be explored by anyone, including beginners. When we need to build a model along the way, we pass our modeling work to “software tutorials” that show how to use high-powered modeling software produced by <a href="http://mmbios.pitt.edu">MMBioS</a> researchers in order to build biological models. The software tutorials allow users wishing to get their hands dirty with modeling software to build all of the models that we need in this course. This allows for a course that can be explored by both casual and serious biological modeling learners alike.</p>

<p>After building a model in a software tutorial, we return to the main text and analyze the results of this model. In this way, the text forms a constant interplay between establishing a biological problem, describing how a model work, implementing that model in a software tutorial, and returning to the text to analyze the model and ask our next question, beginning the cycle anew.</p>

<p>The following contents is a work in progress and will be expanded in the coming weeks. For now, you can find links to the start of each published module below.</p>

<h2 id="prologue-an-introduction-to-biological-modeling-via-random-walks-and-turing-patterns"><a href="../prologue/">Prologue: An introduction to biological modeling via random walks and Turing patterns</a></h2>

<h3 id="main-text">Main text</h3>

<ul>
  <li>
    <p>Introduction: Life is random</p>
  </li>
  <li>
    <p>Alan Turing and the zebra’s stripes</p>
  </li>
  <li>
    <p>An introduction to random walks</p>
  </li>
  <li>
    <p>A reaction-diffusion model generating Turing patterns</p>
  </li>
  <li>
    <p>The Gray-Scott model: a Turing pattern cellular automaton</p>
  </li>
  <li>
    <p>Conclusion: Turing patterns are fine-tuned</p>
  </li>
</ul>

<h3 id="software-tutorials-featuring-mcell-and-cellblender">Software tutorials (featuring MCell and CellBlender)</h3>

<ul>
  <li>
    <p>Simulating particle diffusion with CellBlender</p>
  </li>
  <li>
    <p>Generating Turing patterns with a reaction-diffusion simulation in CellBlender</p>
  </li>
  <li>
    <p>Building a diffusion cellular automaton with Jupyter notebook</p>
  </li>
  <li>
    <p>Implementing the Gray-Scott reaction-diffusion automaton with Jupyter notebook</p>
  </li>
</ul>

<h2 id="module-1-finding-motifs-in-transcription-factor-networks"><a href="../motifs/home">Module 1: Finding motifs in transcription factor networks</a></h2>

<h3 id="main-text-1">Main text</h3>

<ul>
  <li>
    <p>Introduction: Networks rule biology</p>
  </li>
  <li>
    <p>Transcription and DNA-protein binding</p>
  </li>
  <li>
    <p>Transcription factor networks</p>
  </li>
  <li>
    <p>Using randomness to verify network motifs</p>
  </li>
  <li>
    <p>The negative autoregulation motif</p>
  </li>
  <li>
    <p>The feedforward loop motif</p>
  </li>
  <li>
    <p>Building a biological oscillator</p>
  </li>
  <li>
    <p>Conclusion: the importance of robustness in biological oscillations</p>
  </li>
</ul>

<h3 id="software-tutorials-featuring-mcell-and-cellblender-1">Software tutorials (featuring MCell and CellBlender)</h3>

<ul>
  <li>
    <p>Hunting for loops in transcription factor networks</p>
  </li>
  <li>
    <p>Comparing simple regulation to negative autoregulation</p>
  </li>
  <li>
    <p>Ensuring a mathematically controlled simulation for comparing simple regulation to negative autoregulation</p>
  </li>
  <li>
    <p>Implementing the feed-forward loop motif</p>
  </li>
  <li>
    <p>Implementing the repressilator: a biological oscillator</p>
  </li>
  <li>
    <p>Perturbing the repressilator</p>
  </li>
</ul>

<h2 id="module-2-unpacking-e-colis-genius-exploration-algorithm"><a href="../chemotaxis/home">Module 2: Unpacking E. coli’s genius exploration algorithm</a></h2>

<h3 id="main-text-2">Main text</h3>

<ul>
  <li>
    <p>Introduction: The lost immortals</p>
  </li>
  <li>
    <p>Bacterial runs and tumbles</p>
  </li>
  <li>
    <p>Signaling and ligand-receptor dynamics</p>
  </li>
  <li>
    <p>Stochastic simulation of chemical reactions</p>
  </li>
  <li>
    <p>A biochemically accurate model of bacterial chemotaxis</p>
  </li>
  <li>
    <p>Methylation helps a bacterium adapt to differing concentrations</p>
  </li>
  <li>
    <p>Modeling a bacterium’s response to an attractant gradient</p>
  </li>
  <li>
    <p>Conclusion: the beauty of <em>E. coli</em>’s robust randomized exploration algorithm</p>
  </li>
</ul>

<h3 id="software-tutorials-featuring-bionetgen">Software tutorials (featuring BioNetGen)</h3>

<ul>
  <li>
    <p>Getting started wtih BioNetGen and modeling ligand-receptor dynamics</p>
  </li>
  <li>
    <p>Adding phosphorylation to our BioNetGen model</p>
  </li>
  <li>
    <p>Modeling bacterial adaptation to changing attractant</p>
  </li>
  <li>
    <p>Traveling up an attractant gradient</p>
  </li>
  <li>
    <p>Traveling down an attractant gradient</p>
  </li>
  <li>
    <p>Modeling a pure random walk strategy</p>
  </li>
  <li>
    <p>Modeling E. coli’s sophisticated random walk algorithm</p>
  </li>
  <li>
    <p>Comparing different chemotaxis default tumbling frequencies</p>
  </li>
</ul>

<h2 id="module-3-analyzing-the-coronavirus-spike-protein"><a href="../coronavirus/home">Module 3: Analyzing the coronavirus spike protein</a></h2>

<ul>
  <li>Introduction: A tale of two doctors</li>
</ul>

<h3 id="part-1-protein-structure-prediction">Part 1: Protein structure prediction</h3>

<ul>
  <li>
    <p>An introduction to protein structure prediction</p>
  </li>
  <li>
    <p><em>Ab initio</em> protein structure prediction</p>
  </li>
  <li>
    <p>Homology modeling for protein structure prediction</p>
  </li>
  <li>
    <p>Comparing protein structures to assess model accuracy</p>
  </li>
  <li>
    <p>Part 1 conclusion: protein structure prediction is solved! (Kinda…)</p>
  </li>
</ul>

<h3 id="part-2-comparing-sars-cov-2-and-sars-cov">Part 2: Comparing SARS-CoV-2 and SARS-CoV</h3>

<ul>
  <li>
    <p>Searching for local differences in the SARS-CoV and SARS-CoV-2 spike proteins</p>
  </li>
  <li>
    <p>Analyzing structural differences in the bonding of SARS-CoV and SARS-CoV-2 with the ACE2 enzyme</p>
  </li>
  <li>
    <p>Quantifying the interaction energy between the SARS-CoV-2 spike protein and ACE2</p>
  </li>
  <li>
    <p>Part 2 conclusion: from static protein analysis to molecular dynamics</p>
  </li>
</ul>

<h3 id="software-tutorials-featuring-prody">Software tutorials (featuring ProDy)</h3>

<ul>
  <li>
    <p>Using <em>ab initio</em> modeling to predict the structure of hemoglobin subunit alpha</p>
  </li>
  <li>
    <p>Using homology modeling to predict the structure of the SARS-CoV-2 spike protein</p>
  </li>
  <li>
    <p>Using RMSD to compare the predicted SARS-CoV-2 spike protein against its experimentally validated structure</p>
  </li>
  <li>
    <p>Finding local differences in the SARS-CoV and SARS-CoV-2 spike protein structures</p>
  </li>
  <li>
    <p>Visualizing specific regions of interest within the spike protein structure</p>
  </li>
  <li>
    <p>Computing the energy contributed by a local region of the SARS-CoV-2 spike protein bound with the human ACE2 enzyme</p>
  </li>
  <li>
    <p>Molecular dynamics analysis of coronavirus spike proteins using GNM</p>
  </li>
  <li>
    <p>Adding directionality to spike protein GNM simulations using ANM</p>
  </li>
</ul>

<h2 id="module-4-training-a-computer-to-count-white-blood-cells-automatically">Module 4: Training a computer to count white blood cells automatically</h2>

<p>Coming soon!</p>

<p>Featured software: <a href="http://www.cellorganizer.org">CellOrganizer</a></p>

        
      </section>

      <footer class="page__meta">
        
        


        

      </footer>

      

      
    </div>

    
  </article>

  <!-- only show related on a post page when `related: true`
  
  
  -->
</div>

    </div>

    
      <div class="search-content">
        <div class="search-content__inner-wrap"><div class="search-searchbar"></div>
  <div class="search-hits"></div></div>

      </div>
    

    <div id="footer" class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <hr>
  <a href="http://cbd.cmu.edu" target="_blank"><img align = "left" src="../assets/images/CMU_CBD_logos.png" alt="CBD Logo" width="40%" height="40%"></a>
  <a href="http://mmbios.pitt.edu" target="_blank"><img align="right" src="../assets/images/MMBioS_Logo.png" alt="MMBioS Logo" width="50%" height="50%"></a>
</div>

      </footer>
    </div>

    
  <script src="/assets/js/main.min.js"></script>


<!-- Including InstantSearch.js library and styling -->
<script src="https://cdn.jsdelivr.net/npm/instantsearch.js@2.3.3/dist/instantsearch.min.js"></script>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/instantsearch.js@2.3.3/dist/instantsearch.min.css">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/instantsearch.js@2.3.3/dist/instantsearch-theme-algolia.min.css">

<script>
// Instanciating InstantSearch.js with Algolia credentials
const search = instantsearch({
  appId: 'O24616B64D',
  apiKey: '3d5df9a350e2a9a5039314b58ef8c32d',
  indexName: 'biological_modeling',
  searchParameters: {
    restrictSearchableAttributes: [
      'title',
      'content'
    ]
  }
});

const hitTemplate = function(hit) {
  const url = hit.url;
  const title = hit._highlightResult.title.value;
  const content = hit._highlightResult.html.value;

  return `
    <div class="list__item">
      <article class="archive__item" itemscope itemtype="https://schema.org/CreativeWork">
        <h2 class="archive__item-title" itemprop="headline"><a href="${url}">${title}</a></h2>
        <div class="archive__item-excerpt" itemprop="description">${content}</div>
      </article>
    </div>
  `;
}

// Adding searchbar and results widgets
search.addWidget(
  instantsearch.widgets.searchBox({
    container: '.search-searchbar',
    poweredBy: true,
    placeholder: 'Enter your search term...'
  })
);
search.addWidget(
  instantsearch.widgets.hits({
    container: '.search-hits',
    templates: {
      item: hitTemplate,
      empty: 'No results',
    }
  })
);

// Starting the search only when toggle is clicked
$(document).ready(function () {
  $(".search__toggle").on("click", function() {
    if(!search.started) {
      search.start();
    }
  });
});
</script>








  </body>
</html>
