<!doctype html>
<!--
  Minimal Mistakes Jekyll Theme 4.24.0 by Michael Rose
  Copyright 2013-2020 Michael Rose - mademistakes.com | @mmistakes
  Free for personal and commercial use under the MIT license
  https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE
-->
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin _includes/seo.html --><title>Exercises: Coming Soon! - Biological Modeling: A Free Online Course</title>
<meta name="description" content="A free and open online course in biological modeling at multiple scales.">



<meta property="og:type" content="website">
<meta property="og:locale" content="en_US">
<meta property="og:site_name" content="Biological Modeling: A Free Online Course">
<meta property="og:title" content="Exercises: Coming Soon!">
<meta property="og:url" content="/coronavirus/exercises">


  <meta property="og:description" content="A free and open online course in biological modeling at multiple scales.">












<link rel="canonical" href="/coronavirus/exercises">




<script type="application/ld+json">
  {
    "@context": "https://schema.org",

      "@type": "Person",
      "name": null,
      "url": "/"

  }
</script>







<!-- end _includes/seo.html -->


<!-- https://t.co/dKP3o1e -->
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="/assets/css/main.css">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5/css/all.min.css">



<!--SEO tagging -->
<!-- Begin Jekyll SEO tag v2.6.1 -->
<title>Exercises: Coming Soon! | Biological Modeling: A Free Online Course</title>
<meta name="generator" content="Jekyll v3.9.0" />
<meta property="og:title" content="Exercises: Coming Soon!" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="A free and open online course in biological modeling at multiple scales." />
<meta property="og:description" content="A free and open online course in biological modeling at multiple scales." />
<meta property="og:site_name" content="Biological Modeling: A Free Online Course" />
<meta property="og:image" content="/assets/images/SARS_spike_proteins.jpg" />
<script type="application/ld+json">
{"description":"A free and open online course in biological modeling at multiple scales.","@type":"WebPage","url":"/coronavirus/exercises","publisher":{"@type":"Organization","logo":{"@type":"ImageObject","url":"/assets/images/gray_scott_jupyter_3.png"}},"image":"/assets/images/SARS_spike_proteins.jpg","headline":"Exercises: Coming Soon!","@context":"https://schema.org"}</script>
<!-- End Jekyll SEO tag -->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-173522486-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-173522486-1');
</script>

    <!-- start custom head snippets -->

<!-- insert favicons. use https://realfavicongenerator.net/ -->

<!-- end custom head snippets -->

  </head>

  <body class="layout--single">
    <nav class="skip-links">
  <ul>
    <li><a href="#site-nav" class="screen-reader-shortcut">Skip to primary navigation</a></li>
    <li><a href="#main" class="screen-reader-shortcut">Skip to content</a></li>
    <li><a href="#footer" class="screen-reader-shortcut">Skip to footer</a></li>
  </ul>
</nav>

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->

    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
        <a class="site-title" href="/">Biological Modeling: A Free Online Course</a>
        <ul class="visible-links">


                    <li class="masthead__menu-item">
                        <a href="/meet-the-team/">Course Team</a>
                    </li>




                    <li class="masthead__menu-item">
                        <a href="/contact/">Contact Us</a>
                    </li>




                    <li class="masthead__menu-item">
                        <a href="/contents/">Contents</a>
                    </li>




                    <li class="dropdown ">
                        <a class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Take the Course    <i class="fa fa-caret-down fa-sm" aria-hidden="true"></i><span class="caret"></span></a>
                        <ul class="dropdown-content">

                                <li>
                                    <a href="/prologue/">    <b>Prologue:</b> an introduction to biological modeling via random walks and Turing patterns</a>
                                </li>

                                <li>
                                    <a href="/motifs/home">    <b>Module 1:</b> Finding motifs in transcription factor networks</a>
                                </li>

                                <li>
                                    <a href="/chemotaxis/home">    <b>Module 2:</b> Unpacking E. coli's genius exploration algorithm</a>
                                </li>

                                <li>
                                    <a href="/coronavirus/home">    <b>Module 3:</b> Analyzing the coronavirus spike protein</a>
                                </li>

                                <li>
                                    <a href="">    <b>Module 4:</b> coming soon!</a>
                                </li>

                        </ul>
                    </li>


        </ul>

        <button class="search__toggle" type="button">
          <svg class="icon" width="16" height="16" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15.99 16">
            <path d="M15.5,13.12L13.19,10.8a1.69,1.69,0,0,0-1.28-.55l-0.06-.06A6.5,6.5,0,0,0,5.77,0,6.5,6.5,0,0,0,2.46,11.59a6.47,6.47,0,0,0,7.74.26l0.05,0.05a1.65,1.65,0,0,0,.5,1.24l2.38,2.38A1.68,1.68,0,0,0,15.5,13.12ZM6.4,2A4.41,4.41,0,1,1,2,6.4,4.43,4.43,0,0,1,6.4,2Z" transform="translate(-.01)"></path>
          </svg>
        </button>

        <button class="greedy-nav__toggle hidden" type="button">
          <span class="visually-hidden">Toggle Menu</span>
          <div class="navicon"></div>
        </button>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>


    <div class="initial-content">




<div id="main" role="main">

  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>


  <div class="sidebar sticky">











<nav class="nav__list">

  <input id="ac-toc" name="accordion-toc" type="checkbox" />
  <label for="ac-toc">Toggle Menu</label>
  <ul class="nav__items">

      <li>

          <span class="nav__sub-title">Analyzing the coronavirus spike protein</span>



        <ul>

            <li><a href="/coronavirus/home">Introduction: A tale of two doctors<br><br></a></li>

        </ul>

      </li>

      <li>

          <span class="nav__sub-title">Part 1: Protein Structure Prediction</span>



        <ul>

            <li><a href="/coronavirus/structure_intro">An introduction to protein structure prediction<br><br></a></li>

            <li><a href="/coronavirus/ab_initio">Ab initio protein structure prediction<br><br></a></li>

            <li><a href="/coronavirus/homology">Homology modeling for protein structure prediction<br><br></a></li>

            <li><a href="/coronavirus/accuracy">Comparing protein structures to assess model accuracy<br><br></a></li>

            <li><a href="/coronavirus/conclusion_part_1">Part 1 conclusion: protein structure prediction is solved! (Kinda…)<br><br></a></li>

        </ul>

      </li>

      <li>

          <span class="nav__sub-title">Part 2: Comparing SARS-CoV-2 and SARS</span>



        <ul>

            <li><a href="/coronavirus/multiseq">Searching for local differences in the SARS-CoV and SARS-CoV-2 spike proteins<br><br></a></li>

            <li><a href="/coronavirus/structural_diff">Analyzing structural differences in the bonding of SARS-CoV and SARS-CoV-2 with the ACE2 enzyme<br><br></a></li>

            <li><a href="/coronavirus/NAMD">Quantifying the interaction energy between the SARS-CoV-2 spike protein and ACE2<br><br></a></li>

            <li><a href="/coronavirus/conclusion_part_2">Part 2 conclusion: from static protein analysis to molecular dynamics<br><br></a></li>

        </ul>

      </li>

      <li>

          <span class="nav__sub-title">Software Tutorials (featuring ProDy)</span>



        <ul>

            <li><a href="/coronavirus/tutorial_ab_initio">Using ab initio modeling to predict the structure of hemoglobin subunit alpha<br><br></a></li>

            <li><a href="/coronavirus/tutorial_homology">Using homology modeling to predict the structure of the SARS-CoV-2 spike protein<br><br></a></li>

            <li><a href="/coronavirus/tutorial_rmsd">Using RMSD to compare the predicted SARS-CoV-2 spike protein against its experimentally validated structure<br><br></a></li>

            <li><a href="/coronavirus/tutorial_multiseq">Finding local differences in the SARS-CoV and SARS-CoV-2 spike protein structures<br><br></a></li>

            <li><a href="/coronavirus/tutorial_visualization">Visualizing specific regions of interest within the spike protein structure<br><br></a></li>

            <li><a href="/coronavirus/tutorial_NAMD">Computing the energy contributed by a local region of the SARS-CoV-2 spike protein bound with the human ACE2 enzyme<br><br></a></li>

            <li><a href="/coronavirus/tutorial_GNM">Analysis of coronavirus spike proteins using GNM<br><br></a></li>

            <li><a href="/coronavirus/tutorial_ANM">Adding directionality to spike protein GNM simulations using ANM<br><br></a></li>

        </ul>

      </li>

  </ul>
</nav>



  </div>



  <article class="page" itemscope itemtype="https://schema.org/CreativeWork">
    <meta itemprop="headline" content="Exercises: Coming Soon!">




    <div class="page__inner-wrap">

        <header>
          <h1 id="page-title" class="page__title" itemprop="headline">Exercises: Coming Soon!
</h1>

        </header>


      <section class="page__content" itemprop="text">

          <aside class="sidebar__right sticky">
            <nav class="toc">
              <header><h4 class="nav__title"><i class="fas fa-file-alt"></i> On this page</h4></header>


            </nav>
          </aside>

        <ul>
  <li>
    <p>Good exercise: find centroid of a given shape.</p>
  </li>
  <li>
    <p>exercise: compute RMSD.</p>
  </li>
  <li>
    <p>Good exercise later: compute Q scores for the protein structure comparison that we performed at the end of part 1.</p>
  </li>
  <li>
    <p>Good exercise: compute Qres for very simple proteins</p>
  </li>
  <li>
    <p>Exercise based on following excellent observation: In the case of RMSD, I believe that they assign the RMSD of an alignment between a residue and gap to be 0, effectively ignoring it. I believe this is how it is ensured that the two sets have the same number of points (alpha carbons) and also one of the shortcomings of using RMSD. I think for prody, you can set a gap penalty during chain matching,</p>
  </li>
</ul>

<p>“Thus, gap-filled alignments focusing on low RMSDs, while accurate and useful for superposition of structures, are sub-optimal for machine learning as the features of many potentially relevant residues are discarded due to a lack of data in those positions. In most cases, positions with over a certain percentage of aligned residues are considered, with gaps replaced by zeros or by the average of the feature values in that position [22].”</p>

<ul>
  <li>(RMSD with gap. Also included a jupyter notebook of this exercise with the necessary files to the email.) In this exercise, we will see what can happen to RMSD calculations when there is a gap in sequence alignment between two proteins. Let’s use our homology modeling result <em>robetta4</em> (single chain of SARS-CoV-2 Spike) and the associated SARS-CoV-2 Spike model <em>6vxx</em> from the PDB.</li>
</ul>

<p>First, calculate the RMSD between the two models by following the RMSD tutorial and using the chain A to chain A matching (matches[0][0] &amp; matches[0][1]). You should get a RMSD of about 2.5853.</p>

<p>If you followed the tutorial, robetta4 should be under the variable struct1. We will create a new variable, <em>struct3</em>, by taking the sequence of <em>robetta4</em> and deleting a large selection. We can create the variable by using:</p>

<p>struct3 = struct1.select(‘resid 1 to 400 or resid 601 to 20000’)</p>

<p>We use a large value ‘20000’ to ensure that the rest of the protein is captured. Variable struct3 will represent the robetta4 model that has a gap/deletion at residue 400 to 600 (a 200 residue gap). Now we will repeat the RMSD calculation using struct3 instead of struct1. You should get a RMSD of about 2.1927. Is this what you expected?</p>

<p>(There are less residues to compare and deviations to consider, which may have attributed to the decreased RMSD score.)</p>

<ul>
  <li>
    <p>Why are contact maps and cross-correlation maps “symmetric” about the main diagonal?</p>
  </li>
  <li>
    <p>Something on identifying a dynamics difference from a contact map or better cross-correlation in similar proteins.</p>
  </li>
  <li>
    <p>If you have not already done so, try modeling the SARS-CoV-2 S protein or RBD using SWISS-MODEL, Robetta, or GalaxyWEB using the steps in<a href="tutorial_homology" target="blank">Homology Structure Prediction Tutorial</a>. Then, use ProDy to calculate the RMSD between your models and the PDB entries <a href="http://www.rcsb.org/structure/6VXX" target="blank">6vxx</a> for the S protein and <a href="http://www.rcsb.org/structure/6LZG" target="blank">6lzg</a> for the RBD. Did your models perform better than our models?</p>
  </li>
  <li>
    <p>Visualize your best performing model(s) and the corresponding PDB entryl in VMD. If the models are sufficiently similar, try performing a structural alignment using Multiseq and see where in the the structure your predicted models did well.</p>
  </li>
  <li>
    <p>Using VMD, model the SARS-CoV-2 S (<a href="http://www.rcsb.org/structure/6VXX" target="blank">6vxx</a>) protein and SARS S (<a href="https://www.rcsb.org/structure/5X58" target="blank">5x58</a>) protein. Create the graphical representation of glycans and compare the number of glycans between the two proteins. Are they any different? Could this possibly be another reason why SARS-CoV-2 is more infectious than SARS?</p>
  </li>
  <li>
    <p>In our GNM tutorial, we created the contact map using the threshold of 20Å. Try making the contact map of one of the chains of SARS-CoV-2 S protein <a href="http://www.rcsb.org/structure/6VXX">6vxx</a> with different thresholds. Do the maps look different?</p>
  </li>
  <li>
    <p>In this module, we only used homology modeling for large molecules such as the SARS-CoV-2 S protein and the RBD. It would be interesting to directly compare the accuracy of homology modeling and <em>ab initio</em> modeling. Try using one of the three homology modeling softwares to predict the structure of the human hemoglobin subunit (<a href="../_pages/coronavirus/files/Human_Hemoglobin_subunit_alpha_Seq.txt">sequence</a>). After you get your predicted models, try calculating the RMSD using the PDB entry <a href="https://www.rcsb.org/structure/1sI4">1si4</a>. How do they compare to the RMSD from our <em>ab inito</em> (QUARK) models?</p>
  </li>
</ul>


      </section>

      <footer class="page__meta">






      </footer>




    </div>


  </article>

  <!-- only show related on a post page when `related: true`


  -->
</div>

    </div>


      <div class="search-content">
        <div class="search-content__inner-wrap"><div class="search-searchbar"></div>
  <div class="search-hits"></div></div>

      </div>


    <div id="footer" class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <hr>
  <table>
<tbody>
<tr>
<td><a href="http://cbd.cmu.edu" target="_blank"><img src="../assets/images/CMU_CBD_logos.png" alt="CBD Logo"></a></td>
<td><a rel="license" target="_blank" href="http://creativecommons.org/licenses/by-nc-nd/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-nd/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-nc-nd/4.0/">Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International License.</a></td>
<td><a href="http://mmbios.pitt.edu" target="_blank"><img src="../assets/images/MMBioS_Logo.png" alt="MMBioS Logo"></a></td>
</tr>
</tbody>
</table>
</div>

      </footer>
    </div>


  <script src="/assets/js/main.min.js"></script>


<!-- Including InstantSearch.js library and styling -->
<script src="https://cdn.jsdelivr.net/npm/instantsearch.js@2.3.3/dist/instantsearch.min.js"></script>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/instantsearch.js@2.3.3/dist/instantsearch.min.css">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/instantsearch.js@2.3.3/dist/instantsearch-theme-algolia.min.css">

<script>
// Instanciating InstantSearch.js with Algolia credentials
const search = instantsearch({
  appId: 'O24616B64D',
  apiKey: '3d5df9a350e2a9a5039314b58ef8c32d',
  indexName: 'biological_modeling',
  searchParameters: {
    restrictSearchableAttributes: [
      'title',
      'content'
    ]
  }
});

const hitTemplate = function(hit) {
  const url = hit.url;
  const title = hit._highlightResult.title.value;
  const content = hit._highlightResult.html.value;

  return `
    <div class="list__item">
      <article class="archive__item" itemscope itemtype="https://schema.org/CreativeWork">
        <h2 class="archive__item-title" itemprop="headline"><a href="${url}">${title}</a></h2>
        <div class="archive__item-excerpt" itemprop="description">${content}</div>
      </article>
    </div>
  `;
}

// Adding searchbar and results widgets
search.addWidget(
  instantsearch.widgets.searchBox({
    container: '.search-searchbar',
    poweredBy: true,
    placeholder: 'Enter your search term...'
  })
);
search.addWidget(
  instantsearch.widgets.hits({
    container: '.search-hits',
    templates: {
      item: hitTemplate,
      empty: 'No results',
    }
  })
);

// Starting the search only when toggle is clicked
$(document).ready(function () {
  $(".search__toggle").on("click", function() {
    if(!search.started) {
      search.start();
    }
  });
});
</script>








  </body>
</html>
