<!doctype html>
<!--
  Minimal Mistakes Jekyll Theme 4.24.0 by Michael Rose
  Copyright 2013-2020 Michael Rose - mademistakes.com | @mmistakes
  Free for personal and commercial use under the MIT license
  https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE
-->
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin _includes/seo.html --><title>Extra - Biological Modeling: A Free Online Course</title>
<meta name="description" content="A free and open online course in biological modeling at multiple scales.">



<meta property="og:type" content="website">
<meta property="og:locale" content="en_US">
<meta property="og:site_name" content="Biological Modeling: A Free Online Course">
<meta property="og:title" content="Extra">
<meta property="og:url" content="http://localhost:4000/coronavirus/extra">


  <meta property="og:description" content="A free and open online course in biological modeling at multiple scales.">












<link rel="canonical" href="http://localhost:4000/coronavirus/extra">




<script type="application/ld+json">
  {
    "@context": "https://schema.org",
    
      "@type": "Person",
      "name": null,
      "url": "http://localhost:4000/"
    
  }
</script>







<!-- end _includes/seo.html -->


<!-- https://t.co/dKP3o1e -->
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="/assets/css/main.css">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5/css/all.min.css">



<!--SEO tagging -->
<!-- Begin Jekyll SEO tag v2.6.1 -->
<title>Extra | Biological Modeling: A Free Online Course</title>
<meta name="generator" content="Jekyll v3.9.0" />
<meta property="og:title" content="Extra" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="A free and open online course in biological modeling at multiple scales." />
<meta property="og:description" content="A free and open online course in biological modeling at multiple scales." />
<link rel="canonical" href="http://localhost:4000/coronavirus/extra" />
<meta property="og:url" content="http://localhost:4000/coronavirus/extra" />
<meta property="og:site_name" content="Biological Modeling: A Free Online Course" />
<meta property="og:image" content="http://localhost:4000/assets/images/SARS_spike_proteins.jpg" />
<script type="application/ld+json">
{"description":"A free and open online course in biological modeling at multiple scales.","@type":"WebPage","url":"http://localhost:4000/coronavirus/extra","image":"http://localhost:4000/assets/images/SARS_spike_proteins.jpg","headline":"Extra","@context":"https://schema.org"}</script>
<!-- End Jekyll SEO tag -->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-173522486-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-173522486-1');
</script>

    <!-- start custom head snippets -->

<!-- insert favicons. use https://realfavicongenerator.net/ -->

<!-- end custom head snippets -->

  </head>

  <body class="layout--single">
    <nav class="skip-links">
  <ul>
    <li><a href="#site-nav" class="screen-reader-shortcut">Skip to primary navigation</a></li>
    <li><a href="#main" class="screen-reader-shortcut">Skip to content</a></li>
    <li><a href="#footer" class="screen-reader-shortcut">Skip to footer</a></li>
  </ul>
</nav>

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->

    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
        <a class="site-title" href="/">Biological Modeling: A Free Online Course</a>
        <ul class="visible-links">
                
                
                    <li class="masthead__menu-item">
                        <a href="/meet-the-team/">Course Team</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/contact/">Contact Us</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/contents/">Contents</a>
                    </li>
                
            
                
                
                    <li class="dropdown ">
                        <a class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Take the Course    <i class="fa fa-caret-down fa-sm" aria-hidden="true"></i><span class="caret"></span></a>
                        <ul class="dropdown-content">
                            
                                <li>
                                    <a href="/prologue/">    <b>Prologue:</b> an introduction to biological modeling via random walks and Turing patterns</a>
                                </li>
                            
                                <li>
                                    <a href="/motifs/home">    <b>Module 1:</b> Finding motifs in transcription factor networks</a>
                                </li>
                            
                                <li>
                                    <a href="/chemotaxis/home">    <b>Module 2:</b> Unpacking E. coli's genius exploration algorithm</a>
                                </li>
                            
                                <li>
                                    <a href="/coronavirus/home">    <b>Module 3:</b> Analyzing the coronavirus spike protein</a>
                                </li>
                            
                                <li>
                                    <a href="">    <b>Module 4:</b> coming soon!</a>
                                </li>
                            
                        </ul>
                    </li>
                
            
        </ul>
        
        <button class="search__toggle" type="button">
          <svg class="icon" width="16" height="16" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15.99 16">
            <path d="M15.5,13.12L13.19,10.8a1.69,1.69,0,0,0-1.28-.55l-0.06-.06A6.5,6.5,0,0,0,5.77,0,6.5,6.5,0,0,0,2.46,11.59a6.47,6.47,0,0,0,7.74.26l0.05,0.05a1.65,1.65,0,0,0,.5,1.24l2.38,2.38A1.68,1.68,0,0,0,15.5,13.12ZM6.4,2A4.41,4.41,0,1,1,2,6.4,4.43,4.43,0,0,1,6.4,2Z" transform="translate(-.01)"></path>
          </svg>
        </button>
        
        <button class="greedy-nav__toggle hidden" type="button">
          <span class="visually-hidden">Toggle Menu</span>
          <div class="navicon"></div>
        </button>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>


    <div class="initial-content">
      



<div id="main" role="main">

  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>

  
  <div class="sidebar sticky">
  
  
    
      
      
      
      
    
    
      

<nav class="nav__list">
  
  <input id="ac-toc" name="accordion-toc" type="checkbox" />
  <label for="ac-toc">Toggle Menu</label>
  <ul class="nav__items">
    
      <li>
        
          <span class="nav__sub-title">Analyzing the coronavirus spike protein</span>
        

        
        <ul>
          
            <li><a href="/coronavirus/home">Introduction: A tale of two doctors<br><br></a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Part 1: Protein Structure Prediction</span>
        

        
        <ul>
          
            <li><a href="/coronavirus/structure_intro">An introduction to protein structure prediction<br><br></a></li>
          
            <li><a href="/coronavirus/ab_initio">Ab initio protein structure prediction<br><br></a></li>
          
            <li><a href="/coronavirus/homology">Homology modeling for protein structure prediction<br><br></a></li>
          
            <li><a href="/coronavirus/accuracy">Comparing protein structures to assess model accuracy<br><br></a></li>
          
            <li><a href="/coronavirus/conclusion_part_1">Part 1 conclusion: protein structure prediction is solved! (Kinda…)<br><br></a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Part 2: Comparing SARS-CoV-2 and SARS</span>
        

        
        <ul>
          
            <li><a href="/coronavirus/multiseq">Searching for local differences in the SARS-CoV and SARS-CoV-2 spike proteins<br><br></a></li>
          
            <li><a href="/coronavirus/structural_diff">Analyzing structural differences in the bonding of SARS-CoV and SARS-CoV-2 with the ACE2 enzyme<br><br></a></li>
          
            <li><a href="/coronavirus/NAMD">Quantifying the interaction energy between the SARS-CoV-2 spike protein and ACE2<br><br></a></li>
          
            <li><a href="/coronavirus/conclusion_part_2">Part 2 conclusion: from static protein analysis to molecular dynamics<br><br></a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Software Tutorials (featuring ProDy)</span>
        

        
        <ul>
          
            <li><a href="/coronavirus/tutorial_ab_initio">Using ab initio modeling to predict the structure of hemoglobin subunit alpha<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_homology">Using homology modeling to predict the structure of the SARS-CoV-2 spike protein<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_rmsd">Using RMSD to compare the predicted SARS-CoV-2 spike protein against its experimentally validated structure<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_multiseq">Finding local differences in the SARS-CoV and SARS-CoV-2 spike protein structures<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_visualization">Visualizing specific regions of interest within the spike protein structure<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_NAMD">Computing the energy contributed by a local region of the SARS-CoV-2 spike protein bound with the human ACE2 enzyme<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_GNM">Analysis of coronavirus spike proteins using GNM<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_ANM">Adding directionality to spike protein GNM simulations using ANM<br><br></a></li>
          
        </ul>
        
      </li>
    
  </ul>
</nav>

    
  
  </div>



  <article class="page" itemscope itemtype="https://schema.org/CreativeWork">
    <meta itemprop="headline" content="Extra">
    
    
    

    <div class="page__inner-wrap">
      
        <header>
          <h1 id="page-title" class="page__title" itemprop="headline">Extra
</h1>
          
        </header>
      

      <section class="page__content" itemprop="text">
        
          <aside class="sidebar__right sticky">
            <nav class="toc">
              <header><h4 class="nav__title"><i class="fas fa-file-alt"></i> On this page</h4></header>
              <ul class="toc__menu"><li><a href="#part-1">Part 1</a></li><li><a href="#protein-shape-determines-binding-affinity-from-structure-intro-but-possibly-just-delete-this">Protein shape determines binding affinity (from structure intro but possibly just delete this)</a></li><li><a href="#extra-ab-initio">Extra ab initio</a></li><li><a href="#extra--threading">Extra – threading</a></li></ul>

            </nav>
          </aside>
        
        <h2 id="part-1">Part 1</h2>

<ul>
  <li>
    <p>Earlier, point out that no videos exist on protein folding on YouTube because this happens in 1/100 to 1/1000th of a second at a molecular level, so it has not been observed.</p>
  </li>
  <li>
    <p>Earlier, fix issue that RMSD of a short protein is smaller. Better: a single aa can disrupt it, so less likely to have minuscule RMSD with a long protein.</p>
  </li>
</ul>

<h2 id="protein-shape-determines-binding-affinity-from-structure-intro-but-possibly-just-delete-this">Protein shape determines binding affinity (from structure intro but possibly just delete this)</h2>

<p>Now that we understand the importance of shape in determining how proteins interact with molecules in their environment, we will spend some time discussing how these interactions are modeled.</p>

<p>The simplest model of protein interactions is Emil Fischer’s <strong>lock and key</strong> model, which dates back to 1894 [^Fischer]. This model considers a protein that is an <strong>enzyme</strong>, which serves as a catalyst for a reaction involving another molecule called a <strong>substrate</strong>, and we think of the substrate as a key fitting into the enzyme lock. If the substrate does not fit into the active site of an enzyme, then the reaction will not occur.</p>

<p>However, proteins are flexible, a fact that we will return to when we discuss the binding of the coronavirus spike protein to a human enzyme in a later lesson. Because of this flexibility, Daniel Koshland introduced a modified model called the <strong>induced fit</strong> model in 1958.[^Koshland] In this model, the enzyme and substrate may not fit perfectly, nor are they rigid like a lock and key. Rather, the two molecules may fit inexactly, changing shape as they bind to mold together more tightly. That having been said, if an enzyme and substrate’s shape do not match well with each other, then they will not bind. For an overview of the induced fit model, please check out this excellent video from Khan Academy.</p>

<iframe width="640" height="360" src="https://www.youtube-nocookie.com/embed/8lUB2sAQkzw" frameborder="0" allowfullscreen=""></iframe>

<h2 id="extra-ab-initio">Extra ab initio</h2>

<ul>
  <li>
    <p>Models published before crystallography can be found here: <a href="https://www.ssgcid.org/cttdb/molecularmodel_list/?target__icontains=BewuA">SSGCID Models</a></p>
  </li>
  <li>
    <p>From Chris: For more information of how to calculate the energies and the functions for potential energy, click <a href="https://www.ks.uiuc.edu/Research/namd/2.9/ug/node22.html" target="_blank">here</a>.</p>
  </li>
</ul>

<h2 id="extra--threading">Extra – threading</h2>

<ul>
  <li>
    <p>We need to make sure that the specification of the .pdb file type comes back somewhere before we give the results. It may be a perfect place to do so in this lesson.</p>
  </li>
  <li>
    <p>Perhaps something about how <strong>threading</strong> works. Fact is that even if a protein doesn’t have a homologous protein in a database, most proteins will still have a protein of very similar structure.</p>
  </li>
  <li>
    <p>Unfortunately, there are occasions where no identified proteins have notable sequence similarities. The alternative is to use threading, or fold recognition. In this case, rather than comparing the target sequence to sequences in the database, this method compares the target sequence to structures themselves. The biological basis of this method is that in nature, protein structures tend to be highly conservative and unique structural folds are therefore limited.</p>
  </li>
  <li>
    <p>A simple explanation of the general threading algorithm is that structure predictions are created by placing or “threading” each amino acid in the target sequence to template structures from a non-redundant template database, and then assessing how well it fits with some scoring function[^score]. Then, the best-fit templates are used to build the predicted model. The scoring function varies per algorithm, but it typically takes secondary structure compatibilities, gap penalties during alignment, and other terms that depend on amino acids that are bought into contact by the alignment.</p>
  </li>
  <li>
    <p>Each software has its own algorithms and method of assembly, such as how to decide which templates to use, how to use the templates, and how to fill in blurry areas (no good matches with templates). Nevertheless, the three softwares essentially build the models by assembling varying fragments from templates. If you would like to learn more about the intricacies of each software, you can follow these linkes: <a href="https://www.rosettacommons.org/docs/latest/application_documentation/structure_prediction/RosettaCM">Robetta</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3462707/">Galaxy</a>, <a href="https://swissmodel.expasy.org/docs/help">SWISS-MODEL</a>.</p>
  </li>
</ul>

        
      </section>

      <footer class="page__meta">
        
        


        

      </footer>

      

      
    </div>

    
  </article>

  <!-- only show related on a post page when `related: true`
  
  
  -->
</div>

    </div>

    
      <div class="search-content">
        <div class="search-content__inner-wrap"><div class="search-searchbar"></div>
  <div class="search-hits"></div></div>

      </div>
    

    <div id="footer" class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <hr>
  <a href="http://cbd.cmu.edu" target="_blank"><img align = "left" src="../assets/images/CMU_CBD_logos.png" alt="CBD Logo" width="40%" height="40%"></a>
  <a href="http://mmbios.pitt.edu" target="_blank"><img align="right" src="../assets/images/MMBioS_Logo.png" alt="MMBioS Logo" width="50%" height="50%"></a>
</div>

      </footer>
    </div>

    
  <script src="/assets/js/main.min.js"></script>


<!-- Including InstantSearch.js library and styling -->
<script src="https://cdn.jsdelivr.net/npm/instantsearch.js@2.3.3/dist/instantsearch.min.js"></script>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/instantsearch.js@2.3.3/dist/instantsearch.min.css">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/instantsearch.js@2.3.3/dist/instantsearch-theme-algolia.min.css">

<script>
// Instanciating InstantSearch.js with Algolia credentials
const search = instantsearch({
  appId: 'O24616B64D',
  apiKey: '3d5df9a350e2a9a5039314b58ef8c32d',
  indexName: 'biological_modeling',
  searchParameters: {
    restrictSearchableAttributes: [
      'title',
      'content'
    ]
  }
});

const hitTemplate = function(hit) {
  const url = hit.url;
  const title = hit._highlightResult.title.value;
  const content = hit._highlightResult.html.value;

  return `
    <div class="list__item">
      <article class="archive__item" itemscope itemtype="https://schema.org/CreativeWork">
        <h2 class="archive__item-title" itemprop="headline"><a href="${url}">${title}</a></h2>
        <div class="archive__item-excerpt" itemprop="description">${content}</div>
      </article>
    </div>
  `;
}

// Adding searchbar and results widgets
search.addWidget(
  instantsearch.widgets.searchBox({
    container: '.search-searchbar',
    poweredBy: true,
    placeholder: 'Enter your search term...'
  })
);
search.addWidget(
  instantsearch.widgets.hits({
    container: '.search-hits',
    templates: {
      item: hitTemplate,
      empty: 'No results',
    }
  })
);

// Starting the search only when toggle is clicked
$(document).ready(function () {
  $(".search__toggle").on("click", function() {
    if(!search.started) {
      search.start();
    }
  });
});
</script>








  </body>
</html>
