\clearpage\null
\thispagestyle{empty}
\AddToShipoutPictureBG*{% Add picture to current page
  \AtStockLowerLeft{% Add picture to lower-left corner of paper stock
    \includegraphics[width=\stockwidth]{../images/front_cover}}
}
\addtocounter{page}{-1}

\clearpage

% Title page

\clearpage\null
\thispagestyle{empty}
\AddToShipoutPictureBG*{% Add picture to current page
  \AtStockLowerLeft{% Add picture to lower-left corner of paper stock
    \includegraphics[width=\stockwidth]{../images/cover_page}}
}
\addtocounter{page}{-1}

\newpage


\thispagestyle{empty}

% Reverse title page

\phantom{}\vspace{17\baselineskip}

\noindent Copyright ©~2022 by Phillip Compeau. All rights reserved.\\

\noindent This book or any portion thereof may not be reproduced or used in any manner whatsoever without the express written permission of the publisher except for the use of brief quotations in a book review.\\

\noindent Printed in the United States of America\\

\noindent First Printing, 2022\\

\noindent ISBN: TBA\\

\noindent Library of Congress Control Number: TBA\\

\noindent Philomath Press, LLC\\
Pittsburgh, PA
\addtocounter{page}{-1}


\begin{dedication}
\textit{For Sandy Nichols, who taught me that mathematics might give me answers, but that writing would always keep me asking questions.}\\[\baselineskip]
\thispagestyle{empty}
\end{dedication}

\newpage
\phantomsection

% Blank page

% Welcome

\newpage
\phantomsection
\chapter{Welcome!}

\begin{enumerate}
\item Thank you for joining us!
\item This is the print companion to \href{https://biologicalmodeling.org}{\textit{Biological Modeling}} (\href{https://biologicalmodeling.org}{biologicalmodeling.org}), a free and open online course teaching how to model biological systems at multiple scales.
\item Explain how everything works in terms of the tutorials. In the E-book, these tutorials are clickable and will take you directly to the appropriate tutorial at the course website.
\item Explain ``STOP'' questions and how they differ from the exercises that you find at the end of the chapter.
\item Student effort. Students are co-authors here.
\end{enumerate}

From web: 

Have you ever wondered why zebras have stripes? Have you ever wondered how your cells can quickly react to their environment and perform complex tasks without intelligence? Have you ever wondered why the original SARS coronavirus fizzled out but SARS-CoV-2 has spread like wildfire around the planet? Have you ever wondered how algorithms can be trained to “see” cells as well as a human?

What these questions share is that they can start to be answered by modeling biological systems at multiple “scales” of resolution, from the microscopic to the molecular.

In this book, we will build models of biological systems that are relatively simple but nevertheless provide us with deep, fascinating insights into how those systems operate.

\phantom{}\vspace{7.7\baselineskip}

\newpage

\nop{contents}{}
\tableofcontents*
\clearpage
\phantomsection

%Thank you for joining us!  Since its first publication in 2014, this textbook has become a bestseller in the field of computational biology, having been adopted by dozens of universities in twenty countries (\autopageref{sec:university_adoptions}).  Yet we are equally proud that the textbook also powers our popular online courses in bioinformatics, which have been completed by thousands of learners around the world.  We hope that you will sign up for our courses and explore our material online as well.
%
%You can also find lecture videos, PowerPoint slides, and answers to FAQs at the textbook website, \url{http://bioinformaticsalgorithms.org}.
%
%The subtitle of this book is \textit{An Active Learning Approach}.  This is because as you read, you will find a number of active learning components that will help you learn the material interactively and at your own pace.
%
%\begin{enumerate}
%\item \Black{Code Challenges} ask you to implement the algorithms that you will encounter (in any programming language you like).  These code challenges are hosted in the ``Bioinformatics Textbook Track'' location on Rosalind (\url{http://rosalind.info}), a website that will automatically test your implementations.  \marginpar{\centering\vskip-9ex\includegraphics[width = 3.6em]{images/CMYK_ed3/logos/rosalind}}
%\vspace{0.5\baselineskip}
%\item \Blue{Charging Stations} provide additional insights on implementing the algorithms you encounter.  However, we suggest trying to solve a Code Challenge before you visit a Charging Station.\marginpar{\centering\vskip-10.3ex{\includegraphics[width = 3.2em]{images/CMYK_ed3/logos/charging_station_intro}\\ \vskip-8.8ex }}
%%As one of our Coursera students once said, ``read, apply, fail the first attempt, then think carefully about the bigger picture as [you] try to understand why [your] first attempt failed.''
%\vspace{0.5\baselineskip}
%\item \Purple{\textbf{Exercise Breaks}} offer ``just in time'' assessments testing your understanding of a topic before moving to the next one.\marginpar{\centering\vskip-6.2ex\includegraphics[width = 3.2em]{images/CMYK_ed3/logos/exercise}}
%\vspace{0.5\baselineskip}
%\item \Red{STOP and Think} questions invite you to slow down and contemplate the current material before continuing to the next topic.\marginpar{\centering\vskip-6.1ex\includegraphics[width = 3.2em]{images/CMYK_ed3/logos/stop_sign_no_border}}
%\vspace{0.5\baselineskip}
%\item \Orange{Detours} provide extra content that didn't quite fit in the main text.\nop{detour}\marginpar{\centering\vskip-3.3ex\hyperlink{detour}{\includegraphics[width = 6em]{images/CMYK_ed3/logos/detour_blank_no_bg_outline}\hskip-7em\vskip-4.9ex\parbox{7em}{\small\Black{~~~~\textsc{DETOUR}~}}}}
%\vspace{0.5\baselineskip}
%\item \Green{Challenge Problems} ask you to apply what you have learned to real experimental datasets.
%\item \Yellow{FAQs} \marginpar{\centering\vskip-4.6ex\includegraphics[width = 3.6em]{images/CMYK_ed3/logos/faq}} are questions that we encountered many times from the learners who have taken our online courses and the students we have taught using this book in our own offline courses.  When you see the question mark symbol in the margin, it indicates that there is an FAQ on the current subject; answers to FAQs are hosted at the textbook website.
%\end{enumerate}
%
%\noindent Finally, we have tried to make the book as fun to read as possible.  For example, since bioinformatics is still an ever-changing frontier, we decided to implement a Wild West theme for the book. Accordingly, you will notice that each chapter begins with a cartoon image showing the authors entangled in some adventure. What do these cartoons have to do with bioinformatics?  Continue reading to find out!


\newpage
\phantomsection
\chapter{Meet the Team}

\noindent\begin{tabular}{l @{\hskip 3em} l}
\parbox{0.28\textwidth}{\noindent \href{http://compeau.cbd.cmu.edu}{\includegraphics[width=0.28\textwidth]{../images/compeau_phillip_circle}}} & \parbox{0.7\textwidth}{\noindent {\small \href{http://compeau.cbd.cmu.edu}{\textbf{\textsc{Phillip Compeau}}} is the founder of \textit{Biological Modeling}. He is an Associate (Teaching) Professor and the Assistant Department Head in the Computational Biology Department at Carnegie Mellon University, where he directs the undergraduate program and co-directs the precollege program in computational biology. His online education projects, including \href{http://rosalind.info}{Rosalind}, \href{https://programmingforlovers.com}{Programming for Lovers}, and the \href{https://www.coursera.org/specializations/bioinformatics}{Bioinformatics Specialization} on Coursera, have reached over a million people. He is also the co-author of \textit{Bioinformatics Algorithms: An Active Learning Approach}, which has been adopted by 200 instructors in 45 countries.}}\\[22ex]

\parbox{0.28\textwidth}{\noindent \href{http://merterm.github.io}{\includegraphics[width=0.28\textwidth]{../images/inan_mert_circle}}} & \parbox{0.7\textwidth}{\noindent {\small \textbf{\textsc{Mert Inan}} is a computer science Ph.D. student at the University of Pittsburgh and an alum of the M.S. in computational biology program at Carnegie Mellon University. He loves interdisciplinary fields and has been working at the intersection of computation, biology, neuroscience, and machine intelligence. Unlocking the secrets of biology is a pleasure that Mert truly enjoys.}}
\end{tabular}

\noindent\begin{tabular}[m]{l @{\hskip 3em} l}
\parbox{0.28\textwidth}{\noindent \href{https://cbb.yale.edu/people/noah-yann-lee}{\includegraphics[width=0.28\textwidth]{../images/lee_noah_circle}}} & \parbox{0.7\textwidth}{\noindent {\small \textbf{\textsc{Noah Yann Lee}} is a PhD student at Yale University in the Computational Biology and Bioinfornatics program and an alum of the undergraduate program in computational biology at Carnegie Mellon University. From running early-childhood educational tests with the Children’s School at Carnegie Mellon for the Global Learning XPRIZE, to cultivating and sequencing phage genomes with the PhageHunters program, Noah has an appreciation for science from the micro to the macro, physical to the digital. Noah is interested in projects and organizations working with STEM education, and science outreach.}}\\[22ex]

\parbox{0.28\textwidth}{\noindent \href{https://www.linkedin.com/in/chris-lee-b1a020158/}{\includegraphics[width=0.28\textwidth]{../images/lee_chris_circle}}} & \parbox{0.7\textwidth}{\noindent {\small \textbf{\textsc{Chris Lee}} is an associate computational biologist at Dicerna Pharmaceuticals. He is an alum of the M.S. in Computational Biology at Carnegie Mellon University; previously, he was an undergraduate student at Rutgers University and worked as an undergraduate researcher studying hydrothermal vent bacteria, where he graduated magna cum laude with a B.A. in Molecular Biology \& Biochemistry and double minor in Chemistry and Computer Science.}}\\[20ex]

\parbox{0.28\textwidth}{\noindent \href{https://www.linkedin.com/in/shuanger-li-38b300133/}{\includegraphics[width=0.28\textwidth]{../images/li_shuanger_circle}}} & \parbox{0.7\textwidth}{\noindent {\small \textbf{\textsc{Shuanger Li}} is a PhD student at the University of Chicago, and an alum of the M.S. in Computational Biology at Carnegie Mellon, where she worked with Dr. Oana Carja on heritable phenotypic variability. She enjoys modeling and simulation as powerful and fun ways to understand biological systems. She double majored in Environmental Sciences and Microbial Biology at UC Berkeley, where she studied Hawaiian arthropod assemblages, spider behaviors, and remediation bioreactors.}}\\

\end{tabular}

\newpage

%\twitter[http://www.twitter.com/phillipcompeau]
%\linkedin[http://www.linkedin.com/in/phillipcompeau/]{0}

 
\clearpage
\phantomsection
\chapter{Acknowledgments}
\label{chapter:acknowledgments}

The online course from which this book derives, \href{https://biologicalmodeling.org}{\textit{Biological Modeling}}, is a training and dissemination effort for the National Center for Multiscale Modeling of Biological Systems (MMBioS). The development of that course has been supported by the National Institutes of Health (grant ID: P41 GM103712).

We would like to thank everyone who has contributed to the development of MMBioS software, as their work allowed this project to come about. We also would like to thank the other members of our training and dissemination team (Alex Ropelewski, Joe Ayoob, and Rozita Laghaei) as well as the Director of MMBioS, Jim Faeder.

We are grateful to Nicole Matamala, who helped develop tutorials for \autoref{chatper:white_blood_cells}, Yanjing Li, who helped develop exercises for \autoref{chapter:white_blood_cells}, and Velasquez Ebanks and Ulani Qi, who helped work on an early version of \textit{Biological Modeling}. Special thanks to Adam Mihalik, who provided helpful comments on an early draft of this work.

\autoref{chapter:motifs} was in part inspired by Uri Alon’s research and superlative textbook \href{https://www.weizmann.ac.il/mcb/UriAlon/introduction-systems-biology-design-principles-biological-circuits}{\textit{An Introduction to Systems Biology}}, a classic that we strongly recommend if you are interested in a greater discussion of biological network motifs.

Finally, and most importantly, the publication of this book was graciously crowdfunded on \href{https://www.kickstarter.com/projects/phillipcompeau/biological-modeling-a-short-tour?ref=user_menu}{Kickstarter}. As part of the campaign, a number of individuals showed their support above and beyond anything we could have ever imagined. As a token of our eternal gratitude, we are listing them below.
\begin{center}
\tabcolsep=2em
\begin{tabular}{l l}
Rumi Naik & Jill Gunn\\
Nicole Fontanese & Thomas Hutchinson Castro\\
Johnathon Nicolas Roman & Coralline Oz\\
Gillian Hawketts & Robert Lucyshyn\\
Ben Mescher & Jack and Will Haselhorst\\
Dave Lemen & Kevin Mai\\
Emily Tu & Othman Soufan\\
Suraj Nair & Joel Rodriguez Medina\\
Michael Li & Jason Dass\\
Austin Boucinha, OCT & Martin\\
Mark Mammel & Akash Ramachandran\\
Will Townes & Darrell Thomas\\
Sean Leonard & Eric Song\\
Hal Snyder & Jared A.~Davis\\
Kevin Reed & Iridium66\\
Maxwell Shapiro & Jérémie Sartor\\
Jeff Buck & Christopher James Langmead\\
Jacqueline McVeigh & Nikolett Lehel Lassie\\
Kelly Johnson & Jonathan McMenamin-Balano\\
Sassy Momassy & Mary B.~Kennedy\\
William S Owen & HD Welker\\
Wayne Myers & Adam Mihalik\\
Luther Tweneboah Evans & Braden Kartchner\\
Christos Noutsos & Costel Radu\\
Michael Levy & Leeza Sergeeva\\
Genene Hayes & Steven Fines\\
Jonathan M. & Gary L.~Keyfauver, PhD\\
Avinash Seetharamaiah & Steve Ross\\
Laura Climer & Joshua Coleman\\
Jim Reesman & Defne Surujon\\
Nick Connors & Pavel Pevzner\\
\end{tabular}
\end{center}

%
%\noindent This textbook was greatly improved by the efforts of a large number of individuals, to whom we owe a debt of gratitude.
%
%The development team, as well as Laurence Bernstein, Ksenia Krasheninnikova, Max Shen, and Jeffrey Yuan, implemented coding challenges and exercises, rendered figures, helped typeset the text, and offered insightful feedback on the manuscript.
%
%Glenn Tesler provided thorough chapter reviews and even implemented some software to catch errors in the early version of the manuscript!
%
%Robin Betz, Petar Ivanov, James Jensen, and Yu Lin provided insightful comments on the manuscript in its early stages. David Robinson was kind enough to copy edit a few chapters and palliate our punctuation maladies.
%
%Randall Christopher brought to life our crazy ideas for illustrations throughout the book, including the chapter header cartoons and the textbook cover.
%
%Andrey Grigoriev and Max Alekseyev gave advice on the content of \autoref{chapter:replication} and \autoref{chapter:rearrangements}, respectively. Martin Tompa helped us develop the narrative in \autoref{chapter:motifs} by suggesting that we analyze latent tuberculosis infection.
%
%Nikolay Vyahhi led a team composed of Andrey Balandin, Artem Suschev, Aleksey Kladov, and Kirill Shikhanov, who worked hard to support an online, interactive version of this textbook used in our online courses on Coursera.
%
%Our students on Coursera, especially Mark Mammel, Isabel Lupiani, Erika Ram\'irez and Dmitry Kuzminov, found hundreds of typos in our preliminary manuscript.
%
%Mikhail Gelfand, Uri Keich, Hosein Mohimani, Son Pham, and Glenn Tesler advised us on some of the book's ``Open Problems'' and led Massive Open Online Research projects (MOORs) in the first session of our online course.
%
%The Big Data to Knowledge Program at the National Institutes of Health, as well as Howard Hughes Medical Institute and the Russian Ministry of Education and Science generously gave their support for the development of the online courses accompanying this textbook.  The Bioinformatics and Systems Biology Program and the Computer Science \& Engineering Department at the University of California, San Diego provided additional support.
%
%Finally, our families gracefully endured the many long days and nights that we spent poring over manuscripts over the last several years; they helped us preserve our sanity along the way.
%
%\vspace{\baselineskip}
%
%\begin{minipage}{\textwidth}
%\raggedleft\noindent\parbox{0.18\textwidth}{\raggedright\itshape P.\,C. and P.\,P.\\
%May 2018
%}
%\end{minipage}

\newpage \phantom{}
\thispagestyle{empty}